client/src/modules/tags/modal/createUpdate.js

```diff
@@ -20,6 +26,14 @@ function TagsModalController(data, $state, TagsService, Notify, Instance, $rootS
       return false;
     }
 
+    if (typeof (vm.tags) === 'object') {
+      delete vm.tags.iconColor;
+    } else if (Array.isArray(vm.tags)) {
+      vm.tags.forEach(t => {
+        delete t.iconColor;
+      });
```

Comment:
It might be cleaner to just turn this into an array.

```js
// delete all iconColors on all tags
[].concat(vm.tags).forEach(tag => {
  delete tag.iconColor;
});
```